### PR TITLE
Ensure that plugin dependencies are aligned with JFR requirements + Remove the obsolete `databound` demo and test

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Contributes to #344 . After this patch, it will be possible to use custom versions in children packaging POMs.

- ~[ ] Allow using custom version numbers in child packaging POMs~ Reverted in #434 
- ~[ ] Allow setting a Jenkinsfile Runner version via `jfr.version`~ Reverted in #434
- [x] Ensure that plugin dependencies are aligned with JFR requirements
- [x] Remove the obsolete Databound demo

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
So this is also "solves" https://github.com/jenkinsci/jenkinsfile-runner/issues/392 for Custom WAR Packager based flows

**Review by abayer — APPROVED** [review]
Looks good to me.

File: Jenkinsfile [hunk 1]
```diff
@@ -69,21 +70,7 @@ demos['cwp'] = {
...
-                        sh "make clean buildInDocker run"
-                    }
-                }
-            }
-        }
-    }
-}
+} */
 
 parallel(demos)
```
Comment (oleg-nenashev) [inline Jenkinsfile]:
```suggestion
}

parallel(demos)
 */

```

**Comment by oleg-nenashev** [comment]
I will merge it to get the boilerplate development going. As documented in https://github.com/jenkinsci/custom-war-packager/issues/193 , it leads to a Custom WAR Packager flow incompatibility, at least for the development versions.

**Pull request merged**